jacsstorage-services/src/main/java/org/janelia/jacsstorage/service/impl/FileSystemStorageService.java

```diff
@@ -0,0 +1,209 @@
+package org.janelia.jacsstorage.service.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.FileStore;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.janelia.jacsstorage.coreutils.IOStreamUtils;
+import org.janelia.jacsstorage.coreutils.PathUtils;
+import org.janelia.jacsstorage.model.jacsstorage.JADEOptions;
+import org.janelia.jacsstorage.service.ContentAccessParams;
+import org.janelia.jacsstorage.model.jacsstorage.JADEStorageURI;
+import org.janelia.jacsstorage.model.jacsstorage.JacsStorageType;
+import org.janelia.jacsstorage.service.ContentException;
+import org.janelia.jacsstorage.service.ContentNode;
+import org.janelia.jacsstorage.service.ContentStorageService;
+import org.janelia.jacsstorage.service.NoContentFoundException;
+import org.janelia.jacsstorage.service.StorageCapacity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileSystemStorageService implements ContentStorageService {
+
+    private final static Logger LOG = LoggerFactory.getLogger(FileSystemStorageService.class);
+
+    FileSystemStorageService() {
+    }
+
+    @Override
+    public boolean canAccess(String contentLocation) {
+        Path contentPath = Paths.get(contentLocation);
+        if (Files.exists(contentPath)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public ContentNode getObjectNode(String contentLocation) {
+        Path contentPath = Paths.get(contentLocation);
+        if (Files.exists(contentPath) && Files.isRegularFile(contentPath)) {
+            return createContentNode(contentPath);
+        } else {
+            throw new ContentException("Content not found for " + contentLocation);
+        }
+    }
+
+    @Override
+    public List<ContentNode> listContentNodes(String contentLocation, ContentAccessParams filterParams) {
+        return listContentFromPath(getContentPath(contentLocation), filterParams);
+    }
+
+    private Path getContentPath(String contentLocation) {
+        Path contentPath = Paths.get(contentLocation);
+        if (Files.notExists(contentPath)) {
+            return null;
```

Comment:
Why is this null, but the IOException causes an exception? Shouldn't it be FileNotFoundException or something?

Reply:
I don't know what is the best behaviour here. My option was to return an empty list which is in line with 'aws cli ls' does when you provide an invalid prefix